app/controllers/verify/finance_controller.rb

```diff
@@ -7,13 +7,18 @@ class FinanceController < ApplicationController
     before_action :confirm_step_allowed
     before_action :submit_idv_form, only: [:create]
     before_action :submit_idv_job, only: [:create]
+    before_action :refresh_if_not_ready, only: [:show]
```

Comment:
I think I misunderstood. I thought each page was checking for the result before the page loaded, but that check is happening after each step is submitted. I was confused by the `show` action because it implies a GET request, but it's submitting a form. Can you think of a better way to handle this?